pyevmasm/evmasm.py

```diff
@@ -1090,6 +596,293 @@ def assemble_hex(asmcode, pc=0, fork=DEFAULT_FORK):
         return '0x' + hexlify(b''.join([x.bytes for x in asmcode])).decode('ascii')
     return '0x' + hexlify(assemble(asmcode, pc=pc, fork=fork)).decode('ascii')
 
+
+class InstructionTable():
+    """
+    EVM Instruction factory
+        Implements an immutable, iterable instruction LUT that can be indexed by both mnemonic or opcode.
+
+    Example::
+
+        >>> from pyevmasm import instruction_tables
+        >>> instruction_table = instruction_tables['byzantium']
+        >>> instruction_table[0]
+        Instruction(0x0, 'STOP', 0, 0, 0, 0, 'Halts execution.', None, 0)
+        >>> instruction_table['STOP']
+        Instruction(0x0, 'STOP', 0, 0, 0, 0, 'Halts execution.', None, 0)
+        >>> i = instruction_table.__iter__()
+        >>> i.__next__()
+        Instruction(0x0, 'STOP', 0, 0, 0, 0, 'Halts execution.', None, 0)
+        >>> i.__next__()
+        Instruction(0x1, 'ADD', 0, 2, 1, 3, 'Addition operation.', None, 0)
+        >>> i.__next__()
+        Instruction(0x2, 'MUL', 0, 2, 1, 5, 'Multiplication operation.', None, 0)
+        >>> i.__next__()
+        Instruction(0x3, 'SUB', 0, 2, 1, 3, 'Subtraction operation.', None, 0)
+
+    """
+    __slots__ = ('_previous_fork', '_ilist', '_cache', '_name_to_opcode')
+    def __init__(self, *args, **kwargs):
+        self._previous_fork = kwargs.get('previous_fork', None)
+        self._ilist = dict(args[0])
```

Comment:
Is there any good reason to use `*args, **kwargs` here while it could be sth like `instructions, previous_fork=None`?